### DISCUSSION TITLE
is there "non-java" version?

### DISCUSSION THREAD

**Discussion by MatejMagat305** [rust-windowing/android-ndk-rs#393] [General]
well, I know that it is herder to implemented, but I have question, is there something like https://github.com/cnlohr/rawdrawandroid, which can create apk only 45KB big (without java)? fastinating is size as well, but mainly that seems like avoid java complete (which theoretically can avoid GC)..., but this is only question, becouse I know that would be horror rewrite ... and I m not certain whether your native activity use java or not ...

**Comment by MarijnS95** [comment]
There's no Java anywhere in this repository, no "horror rewrite" needed. It's native because... there's no Java :slightly_smiling_face: 

There is however always, and will always be Java on the Android side of the fence, no getting around that - but at least it doesn't ship as part of the APK.

**Reply by MatejMagat305** [reply]
thank, and of course jni call api will be there until will be android  ....
